### PR TITLE
Add pyjwt auth backend

### PR DESCRIPTION
**Pull request by dodget**
This draft PR is an intermediate step to make sure the JWT implementation is on the right path. The implementation includes:

- Adding the [pyjwt](https://pypi.org/project/PyJWT/) dependency
- Adding a `CustomJWTAuthentication` authentication class
- Adding functions in the `media_auth.service` module to handle jwt decoding and validation
- Adding unit tests for all new functions in `media_auth.service`

### CODE REVIEW

**Comment by dodget** [comment]
Add endpoint for auth and user profile update.
Subsequent requests skip previous step assuming its already been done.

**Comment by dodget** [comment]
Updates:
- Added a `update_user` function to `media_auth.views` with a corresponding url to handle user/course updating
- Removed the above workflow from the `CustomJWTAuthentication` authentication class. This reduces DB calls from four to two per request. We are still required to lookup the trusted application secret to validate the JWT, and we have to get the user.
- Tests were added to cover the new auth view

File: media_management_api/media_auth/services.py [hunk 1]
```diff
@@ -16,18 +19,50 @@
 # Token refresh time used when obtaining recent tokens to ensure that the token is good for _at least_ 24 hours
 TOKEN_REFRESH = {"hours": 240 - 24}
 
+
+def get_client_key(header):
+    try:
+        return Application.objects.get(client_id=header['client_id']).client_secret
+    except (KeyError, Application.DoesNotExist):
```
Comment (Tanvez) [inline media_management_api/media_auth/services.py:26]: cool! I never seen this pattern in python

**Review by Tanvez — APPROVED** [review]
LGTM! Awesome work!

**Review by arthurian — APPROVED** [review]
This looks good @dodget. Separating out the authorization step avoids unnecessary DB calls, so existing endpoints only need to verify the JWT and get the user.

File: media_management_api/media_auth/views.py [hunk 2]
```diff
@@ -55,3 +54,18 @@ def check_token(request, access_token):
 def destroy_token(request, access_token):
     services.destroy_token(access_token)
     return HttpResponse("Deleted token:%s" % (access_token))
+
+
+@require_http_methods(["GET"])
```
[the remark this answers is not in the document]
Comment (dodget) [reply media_management_api/media_auth/views.py:59]: I was originally using POST, but some complications turned up in testing. I can't really remember what they were, but its probably worth ironing out. POST would be probably be more idiomatic.

**Review by arthurian — APPROVED** [review]
LGTM @dodget! 👍

**Pull request merged**